internal/compiler-interface/src/main/contraband-java/xsbti/compile/IncOptions.java

```diff
@@ -0,0 +1,274 @@
+     * whether to use.
+     */
+    private boolean useCustomizedFileManager;
+    /**
+     * Determines whether incremental compiler should recompile all dependencies of a file
+     * that contains a macro definition.
+     */
+    private java.util.Optional<Boolean> recompileOnMacroDef;
+    /**
+     * Determines whether optimized approach for invalidating sealed classes/trait is used.
+     * Turning this on may cause undercompilation in case of macros that are based sealed
+     * trait/class children enumeration.
+     */
+    private boolean useOptimizedSealed;
+    /** Determines whether incremental compiler stores apis alongside analysis. */
+    private boolean storeApis;
+    /** Determines whether incremental compilation is enabled. */
+    private boolean enabled;
+    /** Extra options */
+    private java.util.Map<String,String> extra;
+    /** Determines whether to log information on file recompiled due to a transitive macro change */
+    private boolean logRecompileOnMacro;
+    /** External hooks that allows clients e.g. IDEs to interacts with compilation internals */
+    private xsbti.compile.ExternalHooks externalHooks;
+    public IncOptions() {
+        super();
+        transitiveStep = xsbti.compile.IncOptions.defaultTransitiveStep();
+        recompileAllFraction = xsbti.compile.IncOptions.defaultRecompileAllFraction();
+        relationsDebug = xsbti.compile.IncOptions.defaultRelationsDebug();
+        apiDebug = xsbti.compile.IncOptions.defaultApiDebug();
+        apiDiffContextSize = xsbti.compile.IncOptions.defaultApiDiffContextSize();
+        apiDumpDirectory = xsbti.compile.IncOptions.defaultApiDumpDirectory();
+        classfileManagerType = xsbti.compile.IncOptions.defaultClassFileManagerType();
+        useCustomizedFileManager = xsbti.compile.IncOptions.defaultUseOptimizedSealed();
+        recompileOnMacroDef = xsbti.compile.IncOptions.defaultRecompileOnMacroDef();
+        useOptimizedSealed = xsbti.compile.IncOptions.defaultUseOptimizedSealed();
+        storeApis = xsbti.compile.IncOptions.defaultStoreApis();
+        enabled = xsbti.compile.IncOptions.defaultEnabled();
+        extra = xsbti.compile.IncOptions.defaultExtra();
+        logRecompileOnMacro = xsbti.compile.IncOptions.defaultLogRecompileOnMacro();
+        externalHooks = xsbti.compile.IncOptions.defaultExternal();
+    }
+    public IncOptions(int _transitiveStep, double _recompileAllFraction, boolean _relationsDebug, boolean _apiDebug, int _apiDiffContextSize, java.util.Optional<java.io.File> _apiDumpDirectory, java.util.Optional<ClassFileManagerType> _classfileManagerType, boolean _useCustomizedFileManager, java.util.Optional<Boolean> _recompileOnMacroDef, boolean _useOptimizedSealed, boolean _storeApis, boolean _enabled, java.util.Map<String,String> _extra, boolean _logRecompileOnMacro, xsbti.compile.ExternalHooks _externalHooks) {
```

Comment:
Is it possible to generate this constructor as `protected`? Anyone using this shouldn't be, as it will cause guaranteed pain in the future.